source/slang/parser.cpp

```diff
@@ -4368,60 +4470,86 @@ namespace Slang
         return modifier;
     }
 
+
+
     static RefPtr<RefObject> parseLayoutModifier(Parser* parser, void* /*userData*/)
     {
-        Modifiers modifiers;
-        RefPtr<Modifier>* modifierLink = &modifiers.first;
-
-        auto beginMarker = new GLSLLayoutModifierGroupBegin();
-        AddModifier(&modifierLink, beginMarker);
+        ModifierListBuilder listBuilder;
 
+        listBuilder.add(new GLSLLayoutModifierGroupBegin());
+        
         parser->ReadToken(TokenType::LParent);
         while (!AdvanceIfMatch(parser, TokenType::RParent))
         {
             auto nameAndLoc = expectIdentifier(parser);
+            const String& nameText = nameAndLoc.name->text;
 
-            RefPtr<GLSLLayoutModifier> modifier;
+            if (nameText == "binding" ||
+                nameText == "set")
+            {
+                GLSLBindingAttribute* attr = listBuilder.find<GLSLBindingAttribute>();
```

Comment:
Do we need to search the list, or could we just have a local variable:

```c++
GLSLBindingAttribute* bindingAttr = nullptr;
```
and then initialize/add it on demand?